include/aspect/material_model/rheology/drucker_prager.h

```diff
@@ -108,8 +116,8 @@ namespace aspect
           compute_strain_rate_and_derivative (const double stress,
                                               const double pressure,
                                               const unsigned int composition,
-                                              const DruckerPragerParameters p) const;
-
+                                              const std::vector<double> &phase_function_values = std::vector<double>(),
```

Comment:
Why do you need the phase information here? Can you compute the DruckerPragerParameters before you call this function?